src/arctica-greeter.vala

```
@@ -1071,12 +1071,24 @@ public class ArcticaGreeter : Object
         /* Set GTK+ settings */
         debug ("Setting GTK+ settings");
         var settings = Gtk.Settings.get_default ();
-        var value = AGSettings.get_string (AGSettings.KEY_THEME_NAME);
+        var agsettings_ = new AGSettings ();
```

I'd NAK this.

There was a reason why I have implemented it the way it currently is: starting AG in normal mode, we can set `gtk-theme-name` and `gtk-icon-theme-name` at the start to the values provided via gsettings and THEN create the first `AGSetting` object, which will query the "default" GTK (icon) theme (which is what we read from gsettings OR the fallback that GTK went to if the data we set was wrong) and cache it for us. This is how I handled a "faulty" theme name in gsettings, if I remember correctly. 

You're breaking this logic with this change: you're now constructing an `AGSettings` object without having set the (icon) theme name previously, so `AGSettings` will query the actual default GTK (icon) theme, which differs from what we'd like to know.

Fortunately, we don't need any of this, because [`ArcticaGreeter`'s `ready_cb` function](https://github.com/ArcticaProject/arctica-greeter/pull/70/commits/abd6d8d598b7ceb051976439e8527f549716d96d#diff-0de1eca16ff24dff9fb4f820051f18ca30fabcb13a42044d931f1aa5eca8e42dL429) will synchronize the high contrast and big font values on startup, which means that users shouldn't notice anything changing.

Actually testing this again, though, it looks like my original idea doesn't work out too well: if the theme name we provide via `gtk-theme-name` is invalid, GTK will fall back to a different theme (probably the one called "Default" or Adwaita), but sadly not update the `gtk-theme-name` property. I guess we can get rid of the default-theme-name stuff in `src/settings.vala` instead and settle on a hybrid of what we had before and what you tried to do: drop this commit and change the other ones, like I'll explain in different comments.

---

I don't want the greeter to start in some Gtk default theme or the normal theme and then switch over to high-contrast theming immediately after coming up, because AGSettings say so.

In my tests with this branch, I observed AG coming up in normal theme mode and then blend over into high-contrast mode. This causes to much visual noise for visual impaired users, imho.

Also, setting Adwaita first (gtk default theme) and then the normal theme does not make sense here, either. As we obviously have access to all those static values in AGSettings, I don't see any problem using them before AGSettings is fully initialized.

What we can avoid is the pre-mature initialization of "agsettings_". We could also do a static call like`AGSettings.get_boolean(AGSettings.KEY_HIGH_CONTRAST)`.

Basically, the only change needed here is dropping "agsettings_ initialization and usage. IMHO.

---

Okay, that's fair. I concur.

In that case, though, I'd like to move the agsettings object creation to the place where you use it first, including the comment I've written for it.

Additionally, we should drop all the private `default_*theme` variables, it doesn't make sense to keep them. Querying the default themes won't help us. They are hardcoded in GTK, loaded up by default and loading a different theme will just silently do nothing if it's not available. We can't really detect that. We also don't care, because if the strings are invalid, nothing will happen and GTK will stay on its default.

---

ok, go ahead then with this... Thanks!